### DISCUSSION TITLE
How to update  Desuuu to last version Klipper ?

### DISCUSSION THREAD

**Discussion by Witawat** [Desuuuu/klipper#44] [Q&A]
How to update  Desuuu to last version Klipper ?

**Comment by Desuuuu** [comment]
Just like you update the original Klipper. Read their FAQ for the details.

**Reply by Sekenenz** [reply]
So we can directly update the firmware from the fluidd's webui without the risk of overwrite anything?

**Reply by Desuuuu** [reply]
I don't know how fluidd's updater works with forks, so I can't tell you for sure.

**Reply by HcNguyen111** [reply]
> So we can directly update the firmware from the fluidd's webui without the risk of overwrite anything?

(Hi from Reddit!)

I replied to my earlier comment on Reddit, but thought I should also post a reply here for everyone else.

I've worked it out, and you CAN update from Fluidd's Software Updates section AFTER you update your `moonraker.cfg` to allow unofficial repos/branches by adding/updating `enable_repo_debug` under `[update_manager]` to `True`:

My `[update_manager]` section:

    [update_manager]
    channel: dev
    enable_auto_refresh: True
    enable_repo_debug: True
    
Reminder to restart Moonraker/your Pi for the setting change to take effect.

**Reply by Thinkersbluff** [reply]
enable_repo_debug has since been deprecated from Moonraker.

My Mainsail Moonraker Update Manager screen now reports: 
    klipper  Invalid
    dgus-reloaded: dgus-reloaded-221016-0-gbd9fd732

![image](https://user-images.githubusercontent.com/36551518/210296671-67fdd42b-e074-40f3-a743-6e27a2b89210.png)
